Prox/Prox/Utilities/Logger.swift

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+enum LoggerLevel: Int {
+    case debug
+    case info
+    case warn
+    case error
+}
+
+class Logger {
+    private let tag: String
+
+    /// All log levels >= this level will be printed.
+    var level: LoggerLevel
+
+    /// Create a Logger instance that prints messages with the given tag.
+    /// Messages are filtered out if they are less than the given level.
+    init(tag: String, level: LoggerLevel) {
+        self.tag = tag
+        self.level = level
+    }
+
+    /// Create a Logger instance that prints messages with the given tag.
+    /// Assigns the level based on the current build channel.
+    convenience init(tag: String) {
+        let level: LoggerLevel
+        switch AppConstants.BuildChannel {
+        case .Debug:
+            level = .debug
+        case .Release:
+            level = .warn
+        default:
```

Comment:
If we add new build configurations, we'll probably forget to updated the logger. I'd rather enumerate all the BuildChannels so the compile fails rather than forget to set the log levels.